src/smt/proof_manager.cpp

```diff
@@ -32,6 +34,8 @@ PfManager::PfManager(context::UserContext* u, SmtEngine* smte)
           d_pnm.get(), u, "smt::PreprocessProofGenerator")),
       d_pfpp(new ProofPostproccess(d_pnm.get(), smte, d_pppg.get())),
       d_lpfpp(new proof::LeanProofPostprocess(d_pnm.get())),
+      d_vpfppcb(d_pnm.get()),
+      d_vpfpp(new proof::VeriTProofPostprocess(d_pnm.get(),d_vpfppcb)),
```

Comment:
```suggestion
      d_vpfpp(nullptr),
```
Them in the body of the constructor check whether the proof format is the veriT one, in which case you reset `d_vpfpp`. 

Another thing is that I'd like the proof manager to only have a link to the postprocessor. I understand the callback as an attribute is necessary because of how you made the postprocessor extend proof node updater. I now think that it's better not to do it and follow the pattern of the other postprocessors (their own class, whose `process` method uses a local proof node updater).

Reply:
@scottviteri this is something also concerning lean

Reply:
Done